source/module_base/test/mathzone_add1_test.cpp

```diff
@@ -0,0 +1,222 @@
+#include "../mathzone_add1.h"
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+/************************************************
+ *  unit test of class Mathzone_Add1
+ ***********************************************/
+
+/**
+ * - Tested Functions:
+ *   - CubicSplineBoundary1
+ *     - call SplineD2 and Cubic_Spline_Interpolation
+ *     - to interpolate a function using the first
+ *     - kind of boundary condition:f'(0) = f'(n) = 0.0
+ *   - CubicSplineBoundary2
+ *     - call SplineD2 and Cubic_Spline_Interpolation
+ *     - to interpolate a function using the second
+ *     - kind of boundary condition:f''(0) = f''(n) = 0.0
+ *   - UniRadialF
+ *     - call Uni_RadialF to interpolate the radial part
+ *     - of an atomic orbital function, whose discrete r
+ *     - points are uniform.
+ *   - Factorial
+ *     - calculate the factorial of an integer
+ *   - DualFac
+ *     - calculate the double factorial or
+ *     - semifactorial of an integer
+ */
+
+class MathzoneAdd1Test : public testing::Test
+{
+protected:
+	const int MaxInt = 100;
+	int nr_in        = 17;
+	int nr_out       = 161;
+	double *r_in     = new double[nr_in];
+	double *r_out    = new double[nr_out];
+	double *y2       = new double[nr_in];
+	double *psi_in   = new double[nr_in];
+	double *psi_out  = new double[nr_out];
+	double *dpsi     = new double[nr_out];
+	void TearDown()
+	{
+		delete[] r_in;
```

Comment:
`TearDown()` should be the counterpart of `SetUp()`. However, `r_in` is defined as class member, and this usage may cause memory leak in production code. If you wish to do so, `new double[nr_in]` should be within the constructor of  `class MathzoneAdd1Test`, and `delete` in destructor. Here, a better solution is allocating memory in `SetUp()`.